spec/api/queues_spec.cr

```diff
@@ -52,6 +52,34 @@ describe LavinMQ::HTTP::QueuesController do
     ensure
       s.vhosts["/"].delete_queue("stats_q")
     end
+
+    it "should return no persistent message count" do
+      with_channel do |ch|
+        q = ch.queue("stats_q", auto_delete: false, durable: false, exclusive: false)
+        q.publish "m1"
+      end
+      response = get("/api/queues/%2f/stats_q")
+      response.status_code.should eq 200
+      body = JSON.parse(response.body)
+      body["messages"].should eq 1
+      body["messages_persistent"].should eq 0
+    ensure
+      s.vhosts["/"].delete_queue("stats_q")
+    end
+
+    it "should return persistent message count" do
+      with_channel do |ch|
+        q = ch.queue("stats_q", auto_delete: false, durable: true, exclusive: false)
```

Comment:
i mean, before we would loose the queue, we still store the messages persistently? or do we keep all those messages in memory?

Reply:
@kickster97 
As for my understanding, we store all messages to disk (persistently). However using non-durable queue, they will be discarded regardless of deliver_mode* during a restart, all in all, always seen as transient. 

While using durable queue, we consider all messages persistent regardless of deliver_mode*. Contrary to RabbitMQ that need both durable queue and message deliver_mode set to 2. This will make the messages persistent.

*deliver_mode of messages (1 = transient, 2 = persistent)

@jage please correct me if I'm wrong.